lib/browser/api/dialog.js

```diff
@@ -64,203 +44,140 @@ const normalizeAccessKey = (text) => {
+        throw new TypeError(`${name} must be a string`)
+      }
+    }
+
+    // populate fileDialogProperties
     let dialogProperties = 0
     for (const prop in fileDialogProperties) {
       if (properties.includes(prop)) {
         dialogProperties |= fileDialogProperties[prop]
       }
     }
 
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
-      throw new TypeError('Title must be a string')
-    }
-
-    if (buttonLabel == null) {
-      buttonLabel = ''
-    } else if (typeof buttonLabel !== 'string') {
-      throw new TypeError('Button label must be a string')
-    }
-
-    if (defaultPath == null) {
-      defaultPath = ''
-    } else if (typeof defaultPath !== 'string') {
-      throw new TypeError('Default path must be a string')
-    }
-
-    if (filters == null) {
-      filters = []
-    }
-
-    if (message == null) {
-      message = ''
-    } else if (typeof message !== 'string') {
-      throw new TypeError('Message must be a string')
+    // set up callback if one passed
+    let wrappedCallback = null
+    if (typeof callback === 'function') {
+      wrappedCallback = (success, result, bookmarkData) => {
+        return success ? callback(result, bookmarkData) : callback()
+      }
     }
 
-    const wrappedCallback = typeof callback === 'function' ? function (success, result, bookmarkData) {
-      return success ? callback(result, bookmarkData) : callback()
-    } : null
     const settings = {title, buttonLabel, defaultPath, filters, message, securityScopedBookmarks, window}
     settings.properties = dialogProperties
     return binding.showOpenDialog(settings, wrappedCallback)
   },
 
-  showSaveDialog: function (...args) {
+  showSaveDialog: opts => {
     checkAppInitialized()
 
-    let [window, options, callback] = parseArgs(...args)
-
-    if (options == null) {
-      options = {
-        title: 'Save'
-      }
-    }
-
-    let {buttonLabel, defaultPath, filters, title, message, securityScopedBookmarks = false, nameFieldLabel, showsTagField} = options
-
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
-      throw new TypeError('Title must be a string')
-    }
-
-    if (buttonLabel == null) {
-      buttonLabel = ''
-    } else if (typeof buttonLabel !== 'string') {
-      throw new TypeError('Button label must be a string')
-    }
-
-    if (defaultPath == null) {
-      defaultPath = ''
-    } else if (typeof defaultPath !== 'string') {
-      throw new TypeError('Default path must be a string')
-    }
-
-    if (filters == null) {
-      filters = []
-    }
+    // destructure params with default options value
+    let {window, options = {title: 'Save'}, callback} = opts
 
-    if (message == null) {
-      message = ''
-    } else if (typeof message !== 'string') {
-      throw new TypeError('Message must be a string')
-    }
+    // destructure individual options with defaults
+    let {
```

Comment:
`const` maybe? Here and all other places.

Reply:
We wouldn't want to set `const` here because we change some variables later
Ex.
```
 if (options.normalizeAccessKeys) {
      buttons = buttons.map(normalizeAccessKey)
    }
```